### PR TITLE
Attempt to revert resource disablement

### PR DESCRIPTION
**Pull request by deads2k**
An attempt to revert for https://github.com/kubernetes/kubernetes/issues/62044.  It is not clean.

```release-note
Restores ability to enable/disable resource within an API GroupVersion individually.
```

### CODE REVIEW

**Comment by deads2k** [comment]
@kubernetes/sig-api-machinery-bugs  Well, it's now in danger of passing, but it's a dirty revert that actually needs review.

@sttts got appetite to actually try to rewire it just for `kube-apiserver` with less mess than these reverts?

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -147,8 +147,12 @@ func CreateServerChain(runOptions *options.ServerRunOptions, stopCh <-chan struc
 	if err != nil {
 		return nil, err
 	}
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (jennybuckley) [reply cmd/kube-apiserver/app/server.go]: (nevermind it is fixed in a later commit)

**Comment by liggitt** [comment]
/hold
For cleanup

**Comment by deads2k** [comment]
@liggitt thanks, it needs a squash.  Tomorrow am

**Comment by deads2k** [comment]
> @liggitt thanks, it needs a squash. Tomorrow am

and merge conflicts :(

**Comment by deads2k** [comment]
`cmd/kube-apiserver/app/server.go` conflicted heavily.  It'll need another look.

File: cmd/kube-apiserver/app/options/options_test.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ import (
 	apiserveroptions "k8s.io/apiserver/pkg/server/options"
 	genericoptions "k8s.io/apiserver/pkg/server/options"
 	"k8s.io/apiserver/pkg/storage/storagebackend"
-	utilflag "k8s.io/apiserver/pkg/util/flag"
+	utilconfig "k8s.io/apiserver/pkg/util/flag"
```
Comment (sttts) [inline cmd/kube-apiserver/app/options/options_test.go]: strange alias

File: cmd/kube-apiserver/app/server.go [hunk 3]
```diff
@@ -451,7 +448,7 @@ func BuildGenericConfig(
 	kubeVersion := version.Get()
 	genericConfig.Version = &kubeVersion
 
-	storageFactory, lastErr := BuildStorageFactory(s, genericConfig.MergedResourceConfig)
+	storageFactory, err := BuildStorageFactory(s)
```
Comment (sttts) [inline cmd/kube-apiserver/app/server.go]: lastErr

**Comment by deads2k** [comment]
I've got https://github.com/kubernetes/kubernetes/pull/62116 open for 1.10.  I think we should consider merging that for 1.10 and then pursuing an implementation built on what we have (without this revert) in 1.11.  The 1.10 pick was cleaner.

**Comment by hzxuzhonghu** [comment]
@deads2k I would like to.

**Pull request closed without merging**